src/main/java/slimeknights/mantle/recipe/data/BaseRecipeProvider.java

```diff
@@ -0,0 +1,344 @@
+                          .addCriterion("has_item", hasItem(largeItem))
+                          .setGroup(Objects.requireNonNull(smallItem.asItem().getRegistryName()).toString())
+                          .build(consumer, wrap(smallItem, folder, String.format("_from_%s", largeName)));
+  }
+
+  /**
+   * Adds a campfire cooking recipe
+   * @param consumer    Recipe consumer
+   * @param input       Recipe input
+   * @param output      Recipe output
+   * @param experience  Experience for the recipe
+   * @param folder      Folder to store the recipe
+   */
+  protected void addCampfireCooking(Consumer<IFinishedRecipe> consumer, IItemProvider input, IItemProvider output, float experience, String folder) {
+    CookingRecipeBuilder.cookingRecipe(Ingredient.fromItems(input), output, experience, 600, IRecipeSerializer.CAMPFIRE_COOKING)
+            .addCriterion("has_item", hasItem(input))
+            .build(consumer, wrap(output, folder, "_campfire"));
+  }
+
+  /**
+   * Adds a recipe to the campfire, furnace, and smoker
+   * @param consumer    Recipe consumer
+   * @param input       Recipe input
+   * @param output      Recipe output
+   * @param experience  Experience for the recipe
+   * @param folder      Folder to store the recipe
+   */
+  protected void addFoodCooking(Consumer<IFinishedRecipe> consumer, IItemProvider input, IItemProvider output, float experience, String folder) {
+    addCampfireCooking(consumer, input, output, experience, folder);
+    // furnace is 200 ticks
+    ICriterionInstance criteria = hasItem(input);
+    CookingRecipeBuilder.smeltingRecipe(Ingredient.fromItems(input), output, experience, 200)
+            .addCriterion("has_item", criteria)
+            .build(consumer, wrap(output, folder, "_furnace"));
+    // smoker 100 ticks
+    CookingRecipeBuilder.cookingRecipe(Ingredient.fromItems(input), output, experience, 100, IRecipeSerializer.SMOKING)
+            .addCriterion("has_item", criteria)
+            .build(consumer, wrap(output, folder, "_smoker"));
+  }
+
+  /**
+   * Adds recipes to convert a block to ingot, ingot to block, and for nuggets
+   * @param consumer  Recipe consumer
+   * @param metal     Metal object
+   * @param folder    Folder for recipes
+   */
+  protected void registerMineralRecipes(Consumer<IFinishedRecipe> consumer, MetalItemObject metal, String folder) {
+    IItemProvider ingot = metal.getIngot();
+    registerPackingRecipe(consumer, "block", metal.get(), "ingot", ingot, metal.getIngotTag(), folder);
+    registerPackingRecipe(consumer, "ingot", ingot, "nugget", metal.getNugget(), metal.getNuggetTag(), folder);
+  }
+
+  /* conditions */
+
+  /**
+   * Creates a consumer instance with the added conditions
+   * @param consumer    Base consumer
+   * @param conditions  Extra conditions
+   * @return  Wrapped consumer
+   */
+  protected static Consumer<IFinishedRecipe> withCondition(Consumer<IFinishedRecipe> consumer, ICondition... conditions) {
```

Comment:
Compare this to the one from IRecipeBuilderUtils. We may want to merge the two. The utils one stores the consumer in an interface method.